### DISCUSSION TITLE
web question

### DISCUSSION THREAD

**Discussion by ShadiMilad** [community/community#129053] [Programming Help]
### Select Topic Area

Question

### Body

now I have learned web development and refreshed my old information about programming and identifying the new technology in HTML, CSS, and javascript on the web. Still, I am confused about how to apply these in practice and how to begin a project. start with Bootstrap or React etc..., so what's the best practice and order when beginning to design a web?

**Comment by asdacosta** [comment]
Set up your environment with VS Code and GitHub. Define your project scope and build with HTML, style with CSS, and add interactivity with JavaScript. Use Bootstrap for simple projects and React for complex ones. Also test frequently.

**Comment by alwin2024** [comment]
Start by creating small projects such as a weather app you can learn about APIs which opens up a whole new door. You can use html and css to design the web page and javascript for interactivity and to make API calls.

**Comment by danisoaresl** [comment]
Hello.
Plan your website goals and content first. Consider Bootstrap for a quick start or React for more control. Design wireframes, visuals, then code with HTML, CSS, and Javascript. Test, refine, and deploy.

**Comment by shayan-tej** [comment]
You should start creating small web applications and work on your JavaScript skills and do projects and learn new things.

**Reply by ShadiMilad** [reply]
> You should start creating small web applications and work on your JavaScript skills and do projects and learn new things.

start designing app using Bootstrap or React?

**Comment by ebndev** [comment]
Hi @ShadiMilad, thanks for posting in GitHub Discussions! 

I've gone ahead and moved this post to our Programming category, as this topic might be more relevant there! If you could provide more details about what kind of websites or projects you are interested in building, that might help someone give you a nudge in the right direction.

**Reply by ShadiMilad** [reply]
Thanks for responding 
I talk in general any kind of web, e-commerce, educational, organizational, etc ..

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
@ShadiMilad  Brother start with creating small ui design projects like zerodha ui,youtube ui,then apply some js and build projects like calculator,bmi-calculator,clock etc...After that start learning react js and tailwind css..It will be helpful for you

**Comment by manuel14mds** [comment]
I think you have to create or participate in a few projects to apply your knowledge. 
Bootstrap is just for styling, and React is a library/technology for developing websites in a different way. In fact, you could use React with Bootstrap.
